src/cast-player.js

```diff
@@ -877,21 +877,33 @@ class CastPlayer extends BaseRemotePlayer {
         loadOptions.media.breaks = breaks;
       }
     }
-    if (this._playerConfig.sources.captions && this._playerConfig.sources.captions.length)
-      loadOptions.media.tracks = this._playerConfig.sources.captions.map((caption, index) => {
-        let newTrack;
-        newTrack = new chrome.cast.media.Track(index + 1, chrome.cast.media.TrackType.TEXT);
-        Utils.Object.mergeDeep(newTrack, {
-          trackContentId: caption.url,
-          name: caption.label,
-          language: caption.language
-        });
-        return newTrack;
-      });
-
+    const externalCaptions = this._handleExternalCaptions();
+    externalCaptions.length && (loadOptions.media.tracks = externalCaptions);
     return loadOptions;
   }
 
+  _handleExternalCaptions() {
```

Comment:
This method returns the external captions, so `_getExternalCaptions`

Reply:
done